worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageMappingCallable.java

```diff
@@ -52,12 +59,14 @@
     private int missingReferenceYear;
     private int pastReferenceYear;
     private final Map<Integer, Integer> referenceYears = new HashMap<>();
+    private final Ab2dEnvironment appEnv;
 
-    public CoverageMappingCallable(FhirVersion version, CoverageMapping coverageMapping, BFDClient bfdClient) {
+    public CoverageMappingCallable(FhirVersion version, CoverageMapping coverageMapping, BFDClient bfdClient, Ab2dEnvironment appEnv) {
         this.coverageMapping = coverageMapping;
         this.bfdClient = bfdClient;
         this.completed = new AtomicBoolean(false);
-        this.year = getCorrectedYear(coverageMapping.getContract(), coverageMapping.getPeriod().getYear());
+        this.appEnv = appEnv;
```

Comment:
Shouldn't use this. Instead add more to the CoverageMapping object or the ContractForCoverage DTO

Reply:
I was trying to avoid that since this is the only time we check which env is used.

Reply:
We no longer rely on the environment at all for the year. Instead we rely on details about the contract in the database.